packages/dds/map/src/test/mocha/directoryFuzzTests.spec.ts

```diff
@@ -0,0 +1,450 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import * as dirPath from "path";
+import { mkdirSync } from "fs";
+import { strict as assert } from "assert";
+import {
+    BaseFuzzTestState,
+    createFuzzDescribe,
+    createWeightedGenerator,
+    Generator,
+    interleave,
+    makeRandom,
+    performFuzzActions,
+    Reducer,
+    SaveInfo,
+    take,
+} from "@fluid-internal/stochastic-test-utils";
+import {
+    MockContainerRuntimeFactoryForReconnection,
+    MockContainerRuntimeForReconnection,
+    MockFluidDataStoreRuntime,
+    MockStorage,
+} from "@fluidframework/test-runtime-utils";
+import { IChannelServices } from "@fluidframework/datastore-definitions";
+import { DirectoryFactory, SharedDirectory } from "../../directory";
+import { IDirectory } from "../../interfaces";
+
+interface Client {
+    sharedDirectory: SharedDirectory;
+    containerRuntime: MockContainerRuntimeForReconnection;
+}
+
+interface FuzzTestState extends BaseFuzzTestState {
+    containerRuntimeFactory: MockContainerRuntimeFactoryForReconnection;
+    clients: Client[];
+}
+
+interface ClientSpec {
+    directoryId: string;
+}
+
+interface SetKey extends ClientSpec {
+    type: "set";
+    path: string;
+    key: string;
+    value: string;
+}
+
+interface ClearKeys extends ClientSpec {
+    type: "clear";
+    path: string;
+}
+
+interface DeleteKey extends ClientSpec {
+    type: "delete";
+    path: string;
+    key: string;
+}
+
+interface CreateSubDirectory extends ClientSpec {
+    type: "createSubDirectory";
+    path: string;
+    name: string;
+}
+
+interface DeleteSubDirectory extends ClientSpec {
+    type: "deleteSubDirectory";
+    path: string;
+    name: string;
+}
+
+interface Synchronize {
+    type: "synchronize";
+}
+
+type KeyOperation = SetKey | DeleteKey | ClearKeys;
+
+type SubDirectoryOperation = CreateSubDirectory | DeleteSubDirectory;
+
+type ClientOperation = KeyOperation | SubDirectoryOperation;
+
+type Operation = ClientOperation | Synchronize;
+
+interface OperationGenerationConfig {
+    validateInterval: number,
+    maxSubDirectoryChild?: number;
+    subDirectoryNamePool?: string[];
+    keyNamePool?: string[];
+}
+
+const defaultOptions: Required<OperationGenerationConfig> = {
+    validateInterval: 2,
```

Comment:
i see that you override this default in the test below, but still probably worth making it more than 2. That would synchronize after every op.

Reply:
Changed it to 10 too.